lib/puppet/application/lookup.rb

```diff
@@ -339,6 +339,7 @@ def generate_scope
 
     Puppet[:code] = 'undef' unless options[:compile]
     compiler = Puppet::Parser::Compiler.new(node)
+    compiler.loaders # ensures that Pcore et.al. is initialized
```

Comment:
Does this pattern need to be done elsewhere? Thinking of `puppet apply` or other applications that create a `Compiler`?